### PR TITLE
Added documentation for the built-in UCT feature

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
## Purpose of this pull request

This pull request (PR) adds documentation for the Magento 2 PHPStorm plugin built-in upgrade compatibility tool.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/ext-best-practices/phpstorm/introduction.html

## Links to Magento source code

-  https://github.com/magento/magento2-phpstorm-plugin/wiki/4.1.0-Release (need to be reviewed)

whatsnew
Added the [The built-in Upgrade Compatibility Tool (MVP)](https://devdocs.magento.com/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.html) topic.

### CODE REVIEW

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+| 1334 | using @deprecated interface |
+| 1234 | using @deprecated constant |
+| 1534 | using @deprecated property |
+| 1235 | overriding @deprecated constant |
+| 1535 | overriding @deprecated property |
+| 1337 | inherited from @deprecated interface |
+| 1338 | implemented @deprecated interface |
+| 1439 | call @deprecated method |
+
+Complete list of inspections that are already available in the original tool can be found by the [following link]({{site.baseurl}}/upgrade-compatibility-tool/errors.html).
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:26]:
```suggestion
A complete list of inspections that are already available in the original tool, can be found by the [following link]({{site.baseurl}}/upgrade-compatibility-tool/errors.html).
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+This feature can be used in three different ways from the user's point of view:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:28]:
```suggestion
This feature can be used in three different ways, from the user's point of view:
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+*  run compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html) with the familiar output as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:30]:
```suggestion
*  run compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html), with the familiar output, as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+*  run inspections in a real time in the files that are opened in the editor (default)
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:31]:
```suggestion
*  run inspections in real time, on the files that are opened in the editor (default)
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+*  [run inspections manually](https://www.jetbrains.com/help/idea/running-inspections.html) through the user interface on the selected scope of files (default)
+
+It was developed with an aim to use IntelliJ IDEA in most effective way. So, inspections that are running during
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:34]:
```suggestion
It was developed with an aim to use IntelliJ IDEA in the most effective way. So, inspections that are running during
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+the code analysis can alternatively be enabled in the [inspections settings](https://www.jetbrains.com/help/idea/code-inspection.html)
+`Preferences -> Editor -> Inspections -> UCT`:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-settings.png)
+
+It allows you to see the problematic code just in real time without running it intentionally during code writing or even files visiting.
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:40]:
```suggestion
It allows you to see the problematic code in real time, without running it intentionally during code writing or file viewing.
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+**The result of the real time inspection is most suitable for using during development to be always compatible with the future versions**:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:41]:
```suggestion
**The result of the real time inspection is most suitable for use during development, to be always compatible with future versions**:
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-highlighting.png)
+
+The second cool IntelliJ IDEA tool under which we adapted our code by reusing **Inspections** feature of it is running inspections through special action
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:45]: Line 45 - 46 : The English grammar used is very mixed up. Please rewrite in plainer English and use more than one sentence. Omit the word "cool"

I tried to correct the syntax but wasn't totally sure of the meaning of the two sentences.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+`Code -> Inspect Code` or just open context menu under target directory to allow IntelliJ IDEA populates path to analyse for you automatically:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/run-inspect-code-from-context-menu.png)
+
+By calling **Inspect Code** action from the context menu we have Inspection Scope populated for us automatically.
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:50]:
```suggestion
By calling the **Inspect Code** action from the context menu, the Inspection Scope will be populated for us automatically.
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+The last thing to configure here is **Inspection Profile**.
+By default, here will be the **Project Default** profile with all inspections that are not connected to our goal.
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:52]:
```suggestion
By default, there will be the **Project Default** profile with all inspections that are not connected to our goal.
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-intellij-idea-based-analysis-configuration.png)
+
+So you can just create new profile with the only UCT inspections enabled (more about that [read here](https://www.jetbrains.com/help/idea/customizing-profiles.html)):
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:56]:
```suggestion
You can create a new profile with the only UCT inspections enabled (for further information [read here](https://www.jetbrains.com/help/idea/customizing-profiles.html)):
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-profile.png)
+
+For this inspection type you don't need to have UCT inspections enabled for your project.
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:60]:
```suggestion
For this inspection type you do not need to have UCT inspections enabled for your project.
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+**The result of this inspection is most suitable for code refactoring during problematic code elimination**:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:61]:
```suggestion
**The result of this inspection is most suitable for code refactoring when eliminating problematic code**:
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-analysis-through-intellj-idea-inspect-code-action.png)
+
+**The most powerful inspection type is the execution of compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html) with the familiar output as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)**.
+
+**Advantages from using this one:**
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:67]:
```suggestion
**Advantages from using this inspection:**
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+
+*  all problems gathered in one place
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:69]:
```suggestion
*  all problems are gathered in one place
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+*  there are navigations to the problem files, stored report in a json format, link to the web page with detailed description of all the error codes
+*  this report can be easily used from the merchant, management or agencies sides
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:71]:
```suggestion
*  this report can be easily used from the merchant, management or agency side
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+**The result of this inspection is most suitable for code refactoring during problematic code elimination**:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-analysis-through-intellj-idea-inspect-code-action.png)
+
+**The most powerful inspection type is the execution of compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html) with the familiar output as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)**.
+
+**Advantages from using this one:**
+
+*  all problems gathered in one place
+*  there are navigations to the problem files, stored report in a json format, link to the web page with detailed description of all the error codes
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:70]:
```suggestion
*  there are links to the problem files, a stored report in json format, links to the web page with detailed descriptions of all the error codes
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+*  this report can be easily used from the merchant, management or agencies sides
+*  there is a complexity score that helps compare the upgrade complexity from practice side for agencies
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:72]: "practice side" - does not make sense here. Can you explain more clearly?

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+
+Before using it you should configure the built-in UCT in the similar way as the CLI Run Configuration.
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:74]:
```suggestion
Before using it you should configure the built-in UCT in a similar way to the CLI Run Configuration.
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+All fields are described above in the Upgrade Compatibility Tool Run Configuration section and even more detailed
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:75]:
```suggestion
All fields, as above, are described in the Upgrade Compatibility Tool Run Configuration section and are more detailed
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+in the [official documentation]({{site.baseurl}}/upgrade-compatibility-tool/run.html).
+Go to the UCT configuration dialog `Tools -> Configure The Upgrade Compatibility Tool` and configure it as needed:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:77]:
```suggestion
Go to the UCT configuration dialog `Tools -> Configure The Upgrade Compatibility Tool` and configure it as required:
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-settings.png)
+
+Right after that you can run this tool in any time you need that `Tools -> Run The Upgrade Compatibility Tool`:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:81]:
```suggestion
After this, you can run this tool at any time you require by using `Tools -> Run The Upgrade Compatibility Tool`:
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/run-the-built-in-uct-min.gif)
+
+For this inspection type you don't need to have UCT inspections enabled for your project, but it is more comfortable to see visually the problems in the file
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:85]:
```suggestion
For this inspection type you do not need to have UCT inspections enabled for your project, but it is easier to see the problems in the file
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+after navigating to it.
+
+**Keep in mind**, that for all inspection types the target Magento version (coming version) is configured
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:88]:
```suggestion
**Remember**, that for all inspection types the target Magento version (coming version) is configured
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+in the built-in UCT configuration dialog `Tools -> Configure The Upgrade Compatibility Tool`.
+This also applicable for other selected configurations on the screenshot below:
```
Comment (BarnyShergold) [inline src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:90]:
```suggestion
This is also applicable for the other selected configurations on the screenshot below:
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great new technical addition - Mostly grammatical fixes needed.

You have some Linting errors to fix as well -
```
built-in-upgrade-compatibility-tool.md:29: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:30: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:31: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:68: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:69: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:70: MD030 Spaces after list markers
built-in-upgrade-compatibility-tool.md:71: MD030 Spaces after list markers
```

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 25 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:90]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 24 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:88]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 23 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:85]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 22 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:81]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 21 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:77]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 5 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:34]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 4 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:31]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 3 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:30]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 2 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:28]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 1 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:26]: Done.``

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 20 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:75]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 19 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:74]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 18 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:72]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 17 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:70]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 16 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:71]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 15 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:69]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 14 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:67]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 13 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:61]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 12 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:60]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 11 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:56]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 10 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:52]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 9 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:50]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 8 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:45]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 7 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:41]: Done.

File: src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md (hunk 6 above)
Comment (bohdan-harniuk) [reply src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md:40]: Done.

**Comment by bohdan-harniuk** [comment]
Hello, @BarnyShergold! 

Thank you for your review!
I've fixed everything. Please, proceed with the code review.

Thanks, Bohdan.

**Review by danidelcar — APPROVED** [review]

**Review by sivaschenko — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by m2-github-services** [comment]
@dobooth the Pull Request is successfully imported.

**Review by keharper — COMMENTED** [review]
The built-in Upgrade Compatibility Tool (MVP) feels like it came from left field, and I don't understand why it exists or how it's different than the native UCT. Or why it's labeled as MVP.

That said, I'm only flagging some capitalization issues.

**Comment by bohdan-harniuk** [comment]
> The built-in Upgrade Compatibility Tool (MVP) feels like it came from left field, and I don't understand why it exists or how it's different than the native UCT. Or why it's labeled as MVP.
> 
> That said, I'm only flagging some capitalization issues.

Hello, @keharper!

This topic is the part of Magento 2 PHPStorm plugin. The built-in tool is literally built-in into the PHPStorm Plugin. The built-in Upgrade Compatibility Tool aims to achieve the same issues coverage as original Adobe Commerce Upgrade Compatibility Tool. 

For farther information: [watch this](youtube.com/watch?v=6StiTk7oVS8&ab_channel=MagentoCommunityEngineering)

Thanks, Bohdan

**Comment by sivaschenko** [comment]
@keharper the built-in Upgrade Compatibility Tool was designed and implemented in collaboration with the product organization.
"(MVP)" indicates that this is a basic implementation and there are more to come.

**Comment by dshevtsov** [comment]
@keharper,  the link in whatsnew points not to the new topic.

**Comment by dshevtsov** [comment]
I updated the link

**Comment by keharper** [comment]
@bohdan-harniuk The PHPStorm plugin docs were merged while I was on vacation, and I didn't realize this.

**Comment by dshevtsov** [comment]
@dobooth, @keharper, a version label was missing. I'm adding 2.4, because only 2.4 topic is listed in affected pages.

**Pull request merged**